### PR TITLE
Blob Tuning: Revenge of the Buff/Nerf

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
tweak: Resource blobs produce resources for their overmind slightly slower for each resource blob their overmind has.
rscadd: Blob overminds get one free chemical reroll.
rscadd: Blob overminds can no longer place resource and factory blobs out of range of cores and nodes, unless they Toggle Node Requirement off. It is on by default.
rscadd: Blob UI buttons now have tooltips.
tweak: Cores and nodes will expand slightly slower.
tweak: Blob Chemicals no longer trigger effects on dead mobs.
tweak: Shifting Fragments has a lower chance to move shields with normal expansion and will no longer shift blobs to the location of a dying blob.
tweak: Flammable Goo no longer applies fire to tiles with blobs.
tweak: Electromagnetic Web has a slightly higher EMP range.
/:cl:

### CODE REVIEW

**Comment by Zilenan91** [comment]
Thanks for the shifting fragments nerf that blob is cancer

Oh also with blobs no longer applying effects on dead mobs does that still let zombifying feelers work

**Comment by PKPenguin321** [comment]
>rscadd: Blob overminds get one free chemical reroll.

i thought this was removed because blobs would just powergame it and reroll no matter what they got as long as it wasnt the best one

**Comment by ChangelingRain** [comment]
Yeah Zombifying Feelers (and Adaptive Nexuses) only work on unconscious humans

If you can say what the best chem is I'll give you a cookie

**Comment by PKPenguin321** [comment]
>If you can say what the best chem is I'll give you a cookie
 
tru
i guess at the time we did only have like a tenth of the chems we have now

**Comment by Incoming5643** [comment]
>rscadd: Blob overminds can no longer place resource and factory blobs out of range of cores and nodes.

Oh shit it's the revenge of #2791

**Comment by AdamElTablawy** [comment]
Disagree with preventing resource/factory placement outside of range.

Sometimes I build resource blobs if I need that occasional low pulse urgently and don't have enough time to spare for node-resource.

I'd prefer if it was just a warning the first time you did it, as 'This blob will not assist you if placed so far away from a node/core!' and blobs had an option to toggle the warning on to appear every time, or never appear.

**Comment by ChangelingRain** [comment]
Placing resource/factory blobs out of range of nodes/cores used to still have a small chance to still pulse them due to how pulsing worked at the time, but right now it's actively detrimental because pulsing is a non-varying range().

I might make it a toggle, buried deep in the blob's verb menu, but it *will* default to on.

File: code/_onclick/hud/blob_overmind.dm [hunk 1]
```diff
@@ -79,6 +93,14 @@
 	name = "Readapt Chemical (40)"
 	desc = "Randomly rerolls your chemical for 40 points."
 
+/obj/screen/blob/ReadaptChemical/MouseEntered(location,control,params)
+	if(isovermind(usr))
```
Comment (ChangelingRain) [inline code/_onclick/hud/blob_overmind.dm]: short of *really intense* badminning and or shitcoding I'm pretty sure these checks aren't actually needed

File: code/game/gamemodes/blob/overmind.dm [hunk 2]
```diff
@@ -65,8 +68,15 @@
 	..()
 
 /mob/camera/blob/Destroy()
+	for(var/obj/effect/blob/BL in blobs)
```
Comment (RemieRichards) [inline code/game/gamemodes/blob/overmind.dm]: blob list contains blobs soooo

File: code/game/gamemodes/blob/overmind.dm [hunk 3]
```diff
@@ -65,8 +68,15 @@
...
+		if(BL.overmind == src)
+			BL.overmind = null
+			BL.update_icon() //reset anything that was ours
+	for(var/mob/living/simple_animal/hostile/blob/BLO in blob_mobs)
```
Comment (RemieRichards) [inline code/game/gamemodes/blob/overmind.dm]: blob mob list contains blob mobs soooooooo

File: code/game/gamemodes/blob/powers.dm [hunk 4]
```diff
@@ -272,20 +285,20 @@
...
 	var/datum/reagent/blob/BC = pick((subtypesof(/datum/reagent/blob) - blob_reagent_datum.type))
 	blob_reagent_datum = new BC
 	for(var/obj/effect/blob/BL in blobs)
 		BL.update_icon()
-	for(var/mob/living/simple_animal/hostile/blob/BLO)
-		BLO.update_icons()
-		if(BLO.overmind == src) //If it's getting a new chemical, tell it what it does!
-			BLO << "Your overmind's blob reagent is now: <b><font color=\"[blob_reagent_datum.color]\">[blob_reagent_datum.name]</b></font>!"
-			BLO << "The <b><font color=\"[blob_reagent_datum.color]\">[blob_reagent_datum.name]</b></font> reagent [blob_reagent_datum.shortdesc ? "[blob_reagent_datum.shortdesc]" : "[blob_reagent_datum.description]"]"
+	for(var/mob/living/simple_animal/hostile/blob/BLO in blob_mobs)
```
Comment (RemieRichards) [inline code/game/gamemodes/blob/powers.dm]: see above

File: code/modules/reagents/chemistry/reagents/blob_reagents.dm [hunk 5]
```diff
@@ -10,6 +10,8 @@
 	var/message_living = null //extension to first mob sent to only living mobs i.e. silicons have no skin to be burnt
 
 /datum/reagent/blob/reaction_mob(mob/living/M, method=TOUCH, reac_volume, show_message, touch_protection, mob/camera/blob/O)
+	if(M.stat == DEAD)
+		return 0 //the dead don't cause reactions
```
Comment (RemieRichards) [inline code/modules/reagents/chemistry/reagents/blob_reagents.dm]: reagents don't process in the dead

File: code/_onclick/hud/blob_overmind.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/_onclick/hud/blob_overmind.dm]: never forget "AI's laws are a pipe manifold"

File: code/modules/reagents/chemistry/reagents/blob_reagents.dm (hunk 5 above)
Comment (Iamgoofball) [inline code/modules/reagents/chemistry/reagents/blob_reagents.dm]: reaction_mob() still occurs, as that's the "reagents touching the skin"

on_mob_life() is what doesn't happen while dead

**Pull request merged**